src/app/core/ui/search/search.component.ts

```diff
@@ -28,9 +37,91 @@ export class SearchComponent implements OnInit {
 
   ngOnInit() {
     this.createSearchIndex();
+
+    this.searchStringChanged
+      .pipe(debounceTime(this.INPUT_DEBOUNCE_TIME_MS))
+      .subscribe((searchString) => {
+        this.searchQuery.next(this.startSearch(searchString));
+      });
+
+    this.searchQuery
+      .pipe(
+        switchMap((value) => {
+          return value;
+        })
+      )
+      .subscribe(
+        (result: { queryResults: any; searchTerms: string[] }) => {
+          this.handleSearchQueryResult(result);
+        },
+        (error) => {
+          console.error("SearchQuery failed", error);
```

Comment:
Let's use "LoggingService" for this to get it properly registered. Possibly also an error to the user through AlertService?

Reply:
loggingService and alertService added